op-node/rollup/driver/origin_selector.go

```diff
@@ -0,0 +1,69 @@
+package driver
+
+import (
+	"context"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-node/eth"
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+)
+
+type L1Blocks interface {
+	derive.L1BlockRefByHashFetcher
+	derive.L1BlockRefByNumberFetcher
+}
+
+type L1OriginSelector struct {
+	Cfg                 *rollup.Config
```

Comment:
again, fields should tend to be private by default unless the struct is simply a collection of data.

Reply:
Do you prefer `NewXYZ(...)` and private fields? I tend to prefer structs with public fields when the initializer would basically be a pass-through and mutating the fields doesn't really affect the other private fields of the struct.

Reply:
Made the fields private and added `New...` constructors so I can still cover these in action testing